### PR TITLE
ENH: ND States Demo incl PMPS

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Companion to https://github.com/pcdshub/lcls-twincat-motion/pull/163 to include states here.
Also includes the XPIM PMPS demo that I never merged from when we were testing the PMPS database. I think it's important to include it here to make sure I didn't break it (and, so when I update common component library, I can re-use this build for interactive testing)

## Motivation and Context

Make sure the PVs look and behave OK (not accessible fully from the unit tests)

## How Has This Been Tested?

Interactively only

## Where Has This Been Documented?

N/A

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive comments
- [ ] Test suite passes locally
- [x] Libraries are set to ``Always Newest`` version (``Library, *``)
- [x] Committed with ``pre-commit`` or ran ``pre-commit run --all-files``

### CODE REVIEW

File: lcls-plc-example-motion/tc_mot_example/POUs/Main.TcPOU [hunk 1]
```diff
@@ -0,0 +1,339 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1">
+  <POU Name="Main" Id="{7b41e58a-08f0-4959-90b4-af08ef5cbf10}" SpecialFunc="None">
+    <Declaration><![CDATA[PROGRAM Main
+VAR
+    // Simulated motor that is always allowed to move
+    {attribute 'pytmc' := '
+        pv: @PREFIX:SIM:01
```
Comment (klauer) [inline lcls-plc-example-motion/tc_mot_example/POUs/Main.TcPOU:8]: Looks like `@PREFIX` isn't expanded by pytmc; another bug there?

Comment (ZLLentz) [reply lcls-plc-example-motion/tc_mot_example/POUs/Main.TcPOU:8]: Yeah I saw this too- there isn't a conversion specifically for the st.cmd file motor elements

**Comment by ZLLentz** [comment]
This works in conjunction with the example classes in https://github.com/pcdshub/pcdsdevices/pull/1136
```
typhos pcdsdevices.example.PLCExampleMotion[]
```

**Comment by ZLLentz** [comment]
image

**Comment by ZLLentz** [comment]
I also put this into pmps-ui which is neat
```
cd pmps-ui
launch.sh TST
```
with some missing PVs

image

**Comment by ZLLentz** [comment]
I'm going to merge this because it works and I want an example to link to. If there are issues I'll make a new PR.

**Pull request merged**